imports/plugins/core/accounts/client/containers/addressBookContainer.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { compose } from "recompose";
+import { Meteor } from "meteor/meteor";
+import { Template } from "meteor/templating";
+import { i18next } from "/client/api";
+import { Countries } from "/client/collections";
+import * as Collections from "/lib/collections";
+import { registerComponent, composeWithTracker } from "@reactioncommerce/reaction-components";
+import AddressBook from "../components/addressBook";
+
+/**
+ * @method updateAddress
+ * @summary helper function that updates an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {Object} address - address to be updated.
+ * @param{String} property - property to be updated.
+ * @return {Promise}
+ */
+function updateAddress(address, property) {
+  return new Promise((resolve, reject) => {
+    if (property) {
+      Meteor.call("accounts/addressBookUpdate", address, null, property, (error, result) => {
+        if (error || !result) reject(i18next.t("addressBookGrid.somethingWentWrong", { err: error.message }));
+        if (result) resolve(result);
```

Comment:
If for some reason `error` and `result` are ever both truthy, this would end up calling `reject` and `resolve`, which results in an error. It's best to use `else` for all of these situations to ensure it will never call both.

```js
if (error || !result) {
  reject(i18next.t("addressBookGrid.somethingWentWrong", { err: error.message }));
} else {
  resolve(result);
}
```

With promises we always want to make sure exactly one of resolve or reject will be called eventually, down every possible path.